tests/api2/test_300_nfs.py

```diff
@@ -713,32 +844,52 @@ def test_39_check_nfs_service_protocols_parameter(request):
     assert s['nfsd']["vers3"] == 'y', str(s)
     assert s['nfsd']["vers4"] == 'y', str(s)
 
+    # Confirm setting has taken effect: v4->on, v3->on
+    results = SSH_TEST("rpcinfo -s | grep 100003", user, password, ip)
+    assert results['output'].strip().split()[1] == '4,3', results
+
 
 def test_40_check_nfs_service_udp_parameter(request):
     """
     This test verifies that toggling the `udp` option generates expected changes
     in nfs kernel server config.
     """
     depends(request, ["pool_04", "ssh_password"], scope="session")
-
-    results = GET("/nfs")
-    assert results.status_code == 200, results.text
-    assert results.json()['udp'] is False, results.text
-
-    s = parse_server_config()
-    assert s['nfsd']["udp"] == 'n', str(s)
-
-    results = PUT("/nfs/", {"udp": True})
-    assert results.status_code == 200, results.text
-
-    s = parse_server_config()
-    assert s['nfsd']["udp"] == 'y', str(s)
-
-    results = PUT("/nfs/", {"udp": False})
-    assert results.status_code == 200, results.text
-
-    s = parse_server_config()
-    assert s['nfsd']["udp"] == 'n', str(s)
+    with nfs_config():
+        get_payload = {'msg': 'method', 'method': 'nfs.config', 'params': []}
+        set_payload = {'msg': 'method', 'method': 'nfs.update', 'params': []}
+
+        # Initial state should be disabled:
+        #    DB == False, conf == 'n'
+        res = make_ws_request(ip, get_payload)
+        assert res['result']['udp'] is False, res
+        s = parse_server_config()
+        assert s['nfsd']["udp"] == 'n', str(s)
+
+        # Multiple restarts cause systemd failures.  Reset the systemd counters.
+        svcs_to_reset = "nfs-idmapd nfs-mountd nfs-server rpcbind rpc-statd"
+        results = SSH_TEST(f"systemctl reset-failed {svcs_to_reset}", user, password, ip)
+        assert results['result'] is True
```

Comment:
Maybe extract this out for reuse?  I see another call over in _test_015_services.py::test_01_non_silent_service_start_failure_

Reply:
Agreed.

Reply:
done